packages/app/src/OAuthPage.tsx

```diff
@@ -1,20 +1,58 @@
 import { Title } from '@mantine/core';
-import { CodeChallengeMethod } from '@medplum/core';
-import { Logo, SignInForm } from '@medplum/react';
+import { CodeChallengeMethod, normalizeErrorString } from '@medplum/core';
+import { Logo, SignInForm, useMedplum } from '@medplum/react';
 import { useNavigate, useSearchParams } from 'react-router-dom';
 import { getConfig } from './config';
+import { useEffect, useState } from 'react';
+import { showNotification } from '@mantine/notifications';
+
+interface LogoInfo {
+  contentType: string;
+  url: string;
+  title: string;
+}
+
+interface ClientInfo {
+  logo: LogoInfo | null;
+  welcomeString: string | null;
+}
 
 export function OAuthPage(): JSX.Element | null {
   const navigate = useNavigate();
   const [params] = useSearchParams();
-
+  const medplum = useMedplum();
+  const scope = params.get('scope') || 'openid';
+  const [welcomeString, setWelcomeString] = useState<string | null>('Medplum');
+  const [logoUrl, setLogoUrl] = useState<string | null>(null);
   const clientId = params.get('client_id');
+
+  useEffect(() => {
```

Comment:
(Correct me if I'm wrong here) I think this will show a flash of the Medplum logo while the effect is running, yeah?

I think we should probably block rendering of the logo and the welcome string until the client info is loaded.

(Please correct me if that's incorrect).

Reply:
The flash would happen.
Blocking rendering logo and welcome screen.